NREL/SAM
# PV Design UI Improvements

# Pull Request Template

## Description

For configurations with Detailed PV model, make inverter voltage ratings editable on System Design page so users can correct voltage ratings from CEC databsae that don't match manufacturer datasheets.

Improve PV sizing messages with more detailed suggestions:

![image](https://user-images.githubusercontent.com/27711933/218226063-b839ba12-506e-4de9-9ca9-004c980b992b.png)

Make multiple MPPT option more transparent:

![image](https://user-images.githubusercontent.com/27711933/218226163-8d9ef103-e86c-460c-a97b-9eead3ed8d5a.png)

**To test:**

1. Create any Detailed PV or PV Battery case.
2. On system design page, change inverter voltage ratings or number of modules per string, and note improved sizing messages. Test with multiple subarrays.
3. Try changing inverter on Inverter page for different inverter models.
4. Verify MPPT voltage clipping in results, note simulation message if array voltage exceeds inverter Vdc_max.
5. Test with multiple MPPTs.

Fixes #1020

Partially addresses #209

## Type of change

Please delete options that are not relevant.



## Checklist:

If you have added a new compute module in a SSC pull request related to this one, be sure to check the [Process Requirements](https://github.com/NREL/SAM/wiki/Compute-modules-in-SAM).
